pkg/alt/karpenter-core/pkg/operator/operator.go

```diff
@@ -66,14 +76,26 @@ const (
 	component = "controller"
 )
 
+type ccpClient struct{}
+
+func withCCPClient(ctx context.Context, client kubernetes.Interface) context.Context {
+	return context.WithValue(ctx, ccpClient{}, client)
```

Comment:
Do you not need the key to actually be the same object?

I thought you would define a "const" instance of the struct as the ccpCleintKey, and then use it for Set, and Get, no?

Reply:
No, from what I remember it is only the type that matters. @matthchr

Reply:
Ok, saw a few more examples of this, so tried running a test, and doesn't appear to be a difference:
https://go.dev/play/p/vQY7n8VY4C9

Reply:
Did look around online and saw the patterning of:
> type contextKey string
> 
> const ccpCleintKey contextKey = "ccpCleintKey"

However, seems also like a fine usage, and not a bug the way it is. No needed action here, but thought I'd leave what I fount

Reply:
I copied this pattern from knative which does it this way (key is basically a bare struct which I think all compare the same regardless of if it's the same instance or not).

You can see that referenced in the context.WithValue docs too:
https://pkg.go.dev/context#WithValue

> The provided key must be comparable and should not be of type string or any other built-in type to avoid collisions between packages using context. Users of WithValue should define their own types for keys. To avoid allocating when assigning to an interface{}, **context keys often have concrete type struct{}**